src/backend/app/Models/Meter/Meter.php

```diff
@@ -64,8 +65,9 @@ public function accessRate(): AccessRate {
         return $this->tariff->accessRate;
     }
 
-    public function tokens(): HasMany {
-        return $this->hasMany(MeterToken::class);
+    public function tokens(): \Illuminate\Database\Eloquent\Relations\HasManyThrough {
+        return $this->hasManyThrough(Token::class, Device::class, 'device_id', 'device_id', 'id', 'id')
+            ->where('device_type', Meter::class);
```

Comment:
This is great, I wasn't aware we could do this `hasManyThrough`!!! Should we also add this to SHS model?

Reply:
Will add this in #704